src/lib/utils/resolveSearchParameters.ts

```diff
@@ -0,0 +1,16 @@
+import { Index } from '../../widgets/index/index';
+import { SearchParameters } from '../../types';
+
+const resolveSearchParameters = (current: Index): SearchParameters[] => {
+  let parent = current.getParent();
+  let states = [current.getHelper()!.state];
+
+  while (parent !== null) {
```

Comment:
could also have been done recursively and then you wouldn't need the duplicated declaration, but this is easier to follow I guess

Reply:
Yep, I wrote it recursively at first. I think it's easier to follow like this.

```ts
const resolveSearchParameters = (current: Index): SearchParameters[] => {
  const _resolveSearchParameters = (
    state: SearchParameters[],
    parent: Index | null
  ) => {
    if (parent === null) {
      return state;
    }

    return _resolveSearchParameters(
      [parent.getHelper()!.state].concat(state),
      parent.getParent()
    );
  };

  return _resolveSearchParameters(
    [current.getHelper()!.state],
    current.getParent()
  );
};
```